File: be-ojt2026/Services/Inventory/Inventory.Domain/Entities/PurchaseOrder.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Inventory.Domain.Entities;

public partial class PurchaseOrder
{
    public int Id { get; set; }

    public int SupplierId { get; set; }

    public int? CreatedById { get; set; }

    public string? Status { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? ConfirmedAt { get; set; }

    public DateTime? CompletedAt { get; set; }

    public int? ToLocationId { get; set; }
    public DateTime? PlannedDate { get; set; }

    [JsonIgnore]
    public virtual ICollection<PurchaseOrderItem> PurchaseOrderItems { get; set; } = new List<PurchaseOrderItem>();

    public virtual Supplier Supplier { get; set; } = null!;

    public virtual Location? ToLocation { get; set; }
}

File: fe-ojt2026/src/types/warehouse/purchaseOrder.ts
import type { SupplierFromAPI } from "./partners";

export type PurchaseOrderStatus = "Draft" | "Confirmed" | "PartiallyReceived" | "Completed" | "Cancelled";

export interface PurchaseOrderItem {
  Id: number;
  PurchaseOrderId: number;
  ProductId: number;
  ProductName?: string;
  Quantity: number;
  ReceivedQty: number;
  UnitPrice: number;
  Subtotal: number;
  UomId?: number;
  UomName?: string;
  LotId?: number;
  LotNumber?: string;
}

export interface PurchaseOrder {
  Id: number;
  DocumentNo: string;
  SupplierId: number;
  SupplierName?: string;
  Status: PurchaseOrderStatus;
  TotalAmount: number;
  TotalQuantity: number;
  ItemCount: number;
  Notes?: string;
  CreatedAt: string;
  CreatedBy?: number;
  PlannedDate?: string;
  ConfirmedAt?: string;
  CompletedAt?: string;
  Origin?: string;
  ToLocationId?: number;
  ToLocationName?: string;
}

export interface PurchaseOrderListItem extends PurchaseOrder {
  Supplier?: SupplierFromAPI;
}

export interface PurchaseOrderDetail extends PurchaseOrder {
  Items: PurchaseOrderItem[];
}

export interface PurchaseOrderResponse {
  Items: PurchaseOrderListItem[];
  TotalItems: number;
  Page: number;
  PageSize: number;
  TotalPages: number;
  HasNext: boolean;
  HasPrevious: boolean;
}

export interface CreatePurchaseOrderRequest {
  SupplierId: number;
  PlannedDate?: string;
  Notes?: string;
  Items: {
    ProductId: number;
    Quantity: number;
    UnitPrice: number;
    UomId?: number;
  }[];
}

export interface PurchaseOrderSearchParams {
  search?: string;
  status?: string;
  supplierId?: number;
  fromDate?: string;
  toDate?: string;
  fromPlannedDate?: string;
  toPlannedDate?: string;
  page?: number;
  pageSize?: number;
  managerId?: number;
  createdById?: number;
}
